src/main/java/de/feu/massim22/group3/agents/desires/GuardGoalZoneDesire.java

```diff
@@ -0,0 +1,769 @@
+    int stepRecharge = 1;
+    Point gz_point;
+
+    // Set max. Attack distance (distance at which agent will be attacking)
+    final int maxAttackDistance = 1;
+
+    // Flag to indicate if the target GZ was reached once (after that the agent can move more freely)
+    boolean initialReachedGZ = false;
+
+    // State of current explore goal zone (when no enemy with blocks is visible)
+    final String[] directionArray = {"n", "e", "s", "w"};
+    String curDirection = directionArray[0];
+    int curExploreDirection = 0;
+
+    // Information about current targetEnemy
+    Point targetEnemyLastPosition = null;
+    int targetEnemyEnergy = 100;
+    int lastDistToTargetEnemy = 100;
+    boolean lastActionWasClearOnEnemy = false;
+    List<Point> blocksToClear = null;
+
+    // Information about last attacked enemy to avoid attacking again
+    Point lastClearedEnemyPosition = null;
+    int stepsSinceCompleteClear = 0;
+    final int noAttackTime = 10;
+
+    // Positions of all enemies during the previous step
+    List<Point> oldEnemyPositions = new ArrayList<Point>();
+
+    // For path finding
+    Point oldDestPoint = null;
+    Stack<String> dirStack = new Stack<String>();
+
+    public GuardGoalZoneDesire(Belief belief, Point p, String supervisor) {
+        super(belief);
+        gz_point = p;
+        String[] neededActions = {"clear"};
+        precondition.add(new DisconnectAllDesire(belief));
+        precondition.add(new ActionDesire(belief, neededActions));
+    }
+
+    public ActionInfo getNextActionInfo() {
+        ActionInfo a = fulfillPreconditions();
+        if (a != null) {
+            return a;
+        }
+        AgentLogger.info(belief.getAgentShortName() + " GGZD", "");
+        AgentLogger.info(belief.getAgentShortName() + " GGZD",  "##### GGZD new step #####");
+
+
+        // Count successfully cleared blocks
+        if (lastActionWasClearOnBlock) {
+            if (belief.getLastActionResult().equals(ActionResults.SUCCESS)) {
+                clearedBlocks += 1;
+                AgentLogger.info(belief.getAgentShortName() + " GGZD", "Cleared Blocks: " + clearedBlocks);
+            }
+            else {
+                AgentLogger.fine(belief.getAgentShortName() + " GGZD",
+                        "Result of last Clear on block " + belief.getLastActionResult());
+            }
+        lastActionWasClearOnBlock = false;
+        }
+
+
+        // Keep track of energy level of targetEnemy
+        //------------------------------------------
+        if (lastActionWasClearOnEnemy
+                && belief.getLastActionResult().equals(ActionResults.SUCCESS)) {
+            AgentLogger.fine(belief.getAgentShortName() + " GGZD",
+                    "Last clear was a success. Previous energy level: " + targetEnemyEnergy);
+            if (lastDistToTargetEnemy <= clearDamage.length){
+                targetEnemyEnergy = targetEnemyEnergy - clearDamage[lastDistToTargetEnemy] + stepRecharge;
```

Comment:
Es gäbe auch eine server Abfrage über den Zustand des Gegners. Könnte mir vorstellen falls der clear agent erfolgreich arbeitet dass es evtl Sinn macht die Abfrage zu benutzen

Reply:
Meinst du die survey action? Die kann nur der explorer.